web/app/dao/DashboardDAO.java

```diff
@@ -233,314 +266,96 @@
-                                owner.orgHierarchy,
-                                owner.orgHierarchy + "/%");
-
-                        owner.confirmed = getJdbcTemplate().queryForObject(
-                                GET_COUNT_OF_DATASET_WITH_DESCRIPTION,
-                                Long.class,
-                                owner.orgHierarchy,
-                                owner.orgHierarchy + "/%");
-                        if (owner.potentialDatasets != 0)
-                        {
-                            owner.completed = df2.format(100.0 * owner.confirmed / owner.potentialDatasets);
-                        }
-                        else
-                        {
-                            owner.completed = df2.format(0.0);
-                        }
-                        memberList.add(owner);
-                    }
-                }
-            }
+        String totalDatasetQuery;
+        String qualifyingDatasetQuery;
+
+        switch (category) {
+            case Ownership:
+                totalDatasetQuery = GET_COUNT_OF_DATASET_FILTER_PLATFORM;
+                qualifyingDatasetQuery = GET_COUNT_OF_DATASET_WITH_CONFIRMED_OWNER_FILTER_PLATFORM;
+                break;
+
+            case Description:
+                totalDatasetQuery = GET_COUNT_OF_DATASET_FILTER_PLATFORM;
+                qualifyingDatasetQuery = GET_COUNT_OF_DATASET_WITH_DESCRIPTION_FILTER_PLATFORM;
+                break;
+
+            case PrivacyCompliance:
+                totalDatasetQuery = GET_COUNT_OF_DATASET_FILTER_PLATFORM;
+                qualifyingDatasetQuery = GET_COUNT_OF_DATASET_WITH_COMPLIANCE_FILTER_PLATFORM;
+                break;
+
+            case SecuritySpec:
+                totalDatasetQuery = GET_COUNT_OF_DATASET_FILTER_PLATFORM;
+                qualifyingDatasetQuery = GET_COUNT_OF_DATASET_WITH_CONFIDENTIAL_FILTER_PLATFORM;
+                break;
+
+            default:
+                resultNode.put("status", "query failure, unrecognized dashboard category.");
+                return resultNode;
         }
-        resultNode.put("status", "ok");
-        resultNode.set("currentUser", Json.toJson(currentUser));
-        resultNode.set("members", Json.toJson(memberList));
-        return resultNode;
-    }
 
-    public static ObjectNode getIdpcCompliancePercentageByManagerId(String managerId) {
+        DashboardOwnerStat currentUser = null;
+        List<DashboardOwnerStat> memberList = new ArrayList<>();
 
-        ObjectNode resultNode = Json.newObject();
-        ConfidentialFieldsOwner currentUser = null;
-        List<ConfidentialFieldsOwner> memberList = new ArrayList<ConfidentialFieldsOwner>();
-        if (StringUtils.isNotBlank(managerId))
-        {
-            List<LdapInfo> ldapInfoList = JiraDAO.getCurrentUserLdapInfo(managerId);
-            if (ldapInfoList != null && ldapInfoList.size() > 0)
-            {
+        if (StringUtils.isNotBlank(userId)) {
```

Comment:
In general we should use guard clause/early termination where possible to avoid [arrowhead code](https://lostechies.com/chrismissal/2009/05/27/anti-patterns-and-worst-practices-the-arrowhead-anti-pattern/). Let's leave the legacy code alone for now but try not to do this in the future.